### DISCUSSION TITLE
Can libcst be used as python code generation tool?

### DISCUSSION THREAD

**Discussion by kalaskow** [Instagram/LibCST#792] [Q&A] [answered]
Hello,

I've written a couple of codemods to enable large scale refactoring in my project. I've noticed that libcst is a very powerful tool. I know that it could be used for code generation code based libcst classes. My question is: is it recommended to use libcst this way? 

I'm asking because I have a vision to be able to parse a certain DSL and generate python data structures corresponding to data structures in this DSL. Is it recommended to harness libcst to such task or there are more suitable tools/libraries for it?

**Comment by zsol** [comment, marked as answer]
In principle it could be used for code generation but I don't think libcst offers compelling advantages over the builtin ast module + an auto formatter like Black
